### DISCUSSION TITLE
Append a request with a value from a form element?

### DISCUSSION THREAD

**Discussion by MrFaul** [bigskysoftware/htmx#2152] [Q&A] [answered]
Maybe I'm just dumb and there is a super easy solution to this.
My googlefu left me stranded, so I'm here now.
Here is my situation:

Simple API with a URI "thing/number"
Now I have a form with `<input type=number><button hx-get="thing/number from input">` 

Is there a way to do that with htmx?
(Yes, I know, could do that with JS but that's not the point of htmx, isn't it?)
If not, can we get something like `hx-append="selection like hx-include"`?

Changing now the API to accept params would require major PITA.
So I'd like to avoid that.

I think the hx-append attribute idea would be a super powerful client addition.

help pls?

**Comment by Telroshan** [comment, marked as answer]
Hey, I'm not sure to understand your usecase, but could the [`path-params`](https://htmx.org/extensions/path-params/) extension fit your needs here?
Hope it helps!

**Reply by MrFaul** [reply]
Never mind..
This is what I was looking for.
I didn't realize it was an extension.
This should be included in the examples.

**Comment by eydun** [comment]
Maybe this related code can help:

```html
<!DOCTYPE html>
<html lang="en">
<head>
    <script src="https://unpkg.com/htmx.org"></script>
    <script>
        function updateAction() {
            var query = document.getElementById('query').value;
            var form = document.getElementById('searchForm');
            form.setAttribute('hx-get', '/search/' + encodeURIComponent(query));
        }
    </script>
</head>
<body>
    <form id="searchForm" method="get" hx-get="/search" hx-target="#result" hx-trigger="keyup changed delay:500ms">
        <input type="text" id="query" name="query" oninput="updateAction()">
        <input type="submit" value="Search">
    </form>

    <div id="result">
        <!-- Search results will be loaded here -->
    </div>
</body>
</html>
```
